cdm/core/src/main/java/ucar/nc2/ft2/coverage/HorizCoordSys.java

```diff
@@ -646,12 +654,37 @@ private List<LatLonPoint> calcLatLonBoundaryPointsFromProjection(int maxPointsIn
     List<LatLonPoint> latLonPoints = new LinkedList<>();
 
     for (ProjectionPoint projPoint : projPoints) {
-      latLonPoints.add(transform.getProjection().projToLatLon(projPoint));
+      final ProjectionPoint projPointInKm =
+          ProjectionPoint.create(convertToKm(projPoint.getX(), xAxis.units, xAxis.name),
+              convertToKm(projPoint.getY(), yAxis.units, yAxis.name));
+      latLonPoints.add(transform.getProjection().projToLatLon(projPointInKm));
     }
 
     return latLonPoints;
   }
 
+  // TODO is there a better place to handle units?
+  private static double convertToKm(double coordinate, String unit, String axisName) {
+    if (unit.equals("km") || unit.equals("kilometers")) {
+      return coordinate;
+    } else if (unit.equals("m") || unit.equals("meters")) {
+      return 0.001 * coordinate;
+    } else {
+      // TODO not sure what other units that may not need scaling could be here (e.g. radians)
+      return coordinate;
+    }
+  }
+
+  private static double convertFromKm(double coordinate, String unit, String axisName) {
+    if (unit.equals("km") || unit.equals("kilometers")) {
+      return coordinate;
+    } else if (unit.equals("m") || unit.equals("meters")) {
+      return 1000 * coordinate;
+    } else {
```

Comment:
isn't this backwards, i.e. unit=km should return 1000*coordinate?

Reply:
or are the value/units mismatched in this case?

Reply:
Think it is correct but maybe the variable/ function names could be improved-- the coordinate in this case is a computed one in km, and is returned from using the projection to convert from la/lon, i.e. from `proj.latLonToProjBB(llbb)` and the unit is the unit that the data is actually in, e.g. meters.

Maybe `convertFromKm(double coordinateInKm, String desiredUnit, String axisName)` ?